lib/handlers/li.js

```diff
@@ -17,37 +17,43 @@ import {phrasing} from 'hast-util-phrasing'
  *   mdast node.
  */
 export function li(state, node) {
-  const head = node.children[0]
   /** @type {boolean | null} */
   let checked = null
   /** @type {Element | undefined} */
   let clone
 
-  // Check if this node starts with a checkbox.
-  if (head && head.type === 'element' && head.tagName === 'p') {
-    const checkbox = head.children[0]
+  // Check if this node starts with a checkbox (or a paragraph that starts with
+  // a checkbox), indicating the list item is a GFM TaskListItem.
+  let checkboxParent = node
+  let checkbox = node.children[0]
+  if (checkbox && checkbox.type === 'element' && checkbox.tagName === 'p') {
+    checkboxParent = checkbox
+    checkbox = checkbox.children[0]
+  }
 
-    if (
-      checkbox &&
-      checkbox.type === 'element' &&
-      checkbox.tagName === 'input' &&
-      checkbox.properties &&
-      (checkbox.properties.type === 'checkbox' ||
-        checkbox.properties.type === 'radio')
-    ) {
-      checked = Boolean(checkbox.properties.checked)
-      clone = {
-        ...node,
-        children: [
-          {...head, children: head.children.slice(1)},
-          ...node.children.slice(1)
-        ]
-      }
+  if (
+    checkbox &&
+    checkbox.type === 'element' &&
+    checkbox.tagName === 'input' &&
+    checkbox.properties &&
+    (checkbox.properties.type === 'checkbox' ||
+      checkbox.properties.type === 'radio')
+  ) {
```

Comment:
Since Node 16+ is required, this could probably be stremlined a bit with optional chaining (e.g. `checkbox?.properties?.type === 'checkbox'`). I didn't see that in use anywhere else, though, so I left the full check here.